src/libsyntax/ext/mod.rs

```diff
@@ -0,0 +1,25 @@
+//! Definitions shared by macros / syntax extensions and e.g. librustc.
+
+pub mod allocator;
+
+pub mod base {
+    bitflags::bitflags! {
+        /// Built-in derives that need some extra tracking beyond the usual macro functionality.
+        #[derive(Default)]
+        pub struct SpecialDerives: u8 {
+            const PARTIAL_EQ = 1 << 0;
+            const EQ         = 1 << 1;
+            const COPY       = 1 << 2;
+        }
+    }
+}
+
+pub mod proc_macro {
```

Comment:
Both `mod base` and `mod proc_macro` can be inlined into the parent module, IMO.

Reply:
Done.